### DISCUSSION TITLE
How can we set base href during storybook build?

### DISCUSSION THREAD

**Discussion by sonikasharma1403** [nrwl/nx#10735] [General]
How can we set base href during storybook build?

**Comment by rfprod** [comment]
@sonikasharma1403 I know the following approach based on the `browserTarget`

### build target
`targets` -> `build` -> `configuration`
```json
"storybook": {
  "assets": [
    {
      "glob": "**/*.*",
      "input": "./libs/client-core/src/lib/assets",
      "output": "assets"
    }
  ],
  "polyfills": "",
  "scripts": [],
  "styles": [
    "apps/documentation/src/styles.scss",
    "libs/client-core/src/lib/styles/styles.scss",
    "node_modules/roboto-fontface/css/roboto/roboto-fontface.css",
    "node_modules/material-design-icon-fonts-self-hosted-web/iconfont/material-icons.css"
  ],
  "tsConfig": "apps/documentation/.storybook/tsconfig.json"
},
```
example https://github.com/rfprod/nx-ng-starter/blob/main/apps/documentation/project.json#L38C9-L55C10

### build storybook
```json
"build-storybook": {
  "configurations": {
    "ci": {
      "quiet": true
    }
  },
  "executor": "@storybook/angular:build-storybook",
  "options": {
    "browserTarget": "documentation:build:storybook",
    "compodoc": false,
    "configDir": "apps/documentation/.storybook",
    "outputDir": "dist/storybook/documentation"
  }
},
```
example https://github.com/rfprod/nx-ng-starter/blob/main/apps/documentation/project.json#L145C5-L158C7

### start storybook
```json
"storybook": {
  "executor": "@storybook/angular:start-storybook",
  "options": {
    "browserTarget": "documentation:build:storybook",
    "compodoc": false,
    "configDir": "apps/documentation/.storybook",
    "port": 4400
  }
},
```
example https://github.com/rfprod/nx-ng-starter/blob/main/apps/documentation/project.json#L197-L205

The `baseHref` value can be set in the build options https://github.com/rfprod/nx-ng-starter/blob/main/apps/documentation/project.json#L90
However, I'm not sure this has any significant effect.

If one of the stories requires a base href for some reason, maybe a provider can be used https://angular.io/api/common/APP_BASE_HREF

**Reply by eze-nonso** [reply]
@rfprod Could you maybe update the references? As far as I can see from what I try, extra options set like this, such as 'baseHref' are not applied in the final output

**Reply by rfprod** [reply]
@eze-nonso updated.
